lib/cc/analyzer/engine_output_filter.rb

```diff
@@ -0,0 +1,40 @@
+module CC
+  module Analyzer
+    class EngineOutputFilter
+      ISSUE_TYPE = "issue".freeze
+
+      def initialize(config = {})
+        @config = config
+      end
+
+      def filter?(output)
+        if (json = parse_as_json(output))
+          issue?(json) && ignore_issue?(json)
+        else
+          false
+        end
+      end
+
+      private
+
+      def parse_as_json(output)
+        JSON.parse(output)
+      rescue JSON::ParserError
+        nil
+      end
+
+      def issue?(json)
+        json.is_a?(Hash) && json["type"] == ISSUE_TYPE
```

Comment:
If the output is a JSON array (which is valid for JSON). See [here](https://github.com/codeclimate/codeclimate/pull/65/files#diff-1a8dd8bd9de2daee94952f7f6d88199eR15) for a test that fails without this condition in place.